sw/host/opentitanlib/src/transport/ti50emulator/mod.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use log::{error, info};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::fs;
+use std::path::PathBuf;
+use std::process;
+use std::rc::Rc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use crate::io::emu::Emulator;
+use crate::io::gpio::GpioPin;
+use crate::io::i2c::Bus;
+use crate::io::spi::Target;
+use crate::io::uart::Uart;
+use crate::transport::{
+    Capabilities, Capability, Result, Transport, TransportError, TransportInterfaceType,
+};
+
+mod emu;
+mod gpio;
+mod i2c;
+mod spi;
+mod uart;
+
+use crate::transport::ti50emulator::emu::Ti50SubProcess;
+
+pub struct Ti50Emulator {
+    inner: Rc<RefCell<Inner>>,
+}
+
+impl Ti50Emulator {
+    /// Create new instance of [`Ti50Emulator`] based on provided parameters.
+    pub fn open(
+        _executable_directory: PathBuf,
+        _executable: &String,
+        instance_prefix: &String,
+    ) -> anyhow::Result<Self> {
+        let tstamp = SystemTime::now().duration_since(UNIX_EPOCH)?;
+        let instance_name = format!(
+            "{}_{}_{}_{}",
+            instance_prefix,
+            process::id(),
+            tstamp.as_secs(),
+            tstamp.as_nanos()
+        );
+
+        let mut instance_directory = PathBuf::from("/tmp");
+        instance_directory.push(&instance_name);
+        let runtime_directory = instance_directory.join("runtime");
+
+        info!("Initializing Ti50Emulator instance: {}", instance_name);
```

Comment:
A minor style-ism: I prefer logging to be written with the full crate path:
```
log::info!("Initializing the thing");
```

(You can also eliminate the `use log::{...}` statement at the top of this file)

Reply:
I replaced this expression with log::info!(...) in mod.rs.

btw: It look like Verilator backend don't follow this style/rule. 
(5 time in transport.rs and 1 in subprocess.rs)